### PR TITLE
163 semantic diff end to end new

### PR DESCRIPTION
**Pull request by ShiriMoran**

### CODE REVIEW

File: cmd/analyzer/main.go [hunk 1]
```diff
@@ -81,30 +109,61 @@ func _main(cmdlineArgs []string) error {
...
+		var rc2ndForDiff *ibmvpc.ResourcesContainer
+		rc2ndForDiff, err = ibmvpc.ParseResourcesFromFile(*inArgs.InputSecondConfigFile)
+		if err != nil {
+			return fmt.Errorf(ErrorFormat, ParsingErr, err)
+		}
+		vpc2ndConfigs, err := ibmvpc.VPCConfigsFromResources(rc2ndForDiff, *inArgs.VPC, *inArgs.Debug)
+		if err != nil {
+			return fmt.Errorf(ErrorFormat, InGenerationErr, err)
+		}
+		// For diff analysis each vpcConfigs have a single element
```
Comment (adisos) [inline cmd/analyzer/main.go:151]: add a warning/error if this assumption does not hold for now

File: pkg/ibmvpc/analysis_output_test.go [hunk 2]
```diff
@@ -326,6 +330,11 @@ var tests = []*vpcGeneralTest{
 		useCases: []vpcmodel.OutputUseCase{vpcmodel.AllSubnets},
 		format:   vpcmodel.Text,
 	},
+	{
+		name:     "acl_testing5",
+		useCases: []vpcmodel.OutputUseCase{vpcmodel.AllSubnetsDiff},
```
[the remark this answers is not in the document]
Comment (ShiriMoran) [reply pkg/ibmvpc/analysis_output_test.go:340]: Indeed, good catch!

File: cmd/analyzer/main.go [hunk 3]
```diff
@@ -81,33 +109,74 @@ func _main(cmdlineArgs []string) error {
...
+	if err2 != nil {
+		return err2
+	}
+	fmt.Println(analysisOutput.Output)
 	return nil
 }
 
+func getSingleCfg(vpcConfigs map[string]*vpcmodel.VPCConfig) (*vpcmodel.VPCConfig, bool) {
+	if len(vpcConfigs) > 1 {
+		return nil, true
```
Comment (adisos) [inline cmd/analyzer/main.go:172]:
```suggestion
		return nil, false
```

Comment (ShiriMoran) [reply cmd/analyzer/main.go:172]: of course... type

**Review by adisos — APPROVED** [review]

**Pull request merged**